null_safety_examples/misc/bin/named_constructor.dart

```diff
@@ -0,0 +1,20 @@
+// ignore_for_file: sort_constructors_first
+
+// #docregion
+class Point {
+  late double x, y;
+
+  Point(this.x, this.y);
+
+  Point.origin() {
```

Comment:
Based off the comments in the codelab this seems to also be used to intentionally show of the functionality of `late`.

If the desire to illustrate the functionality of `late` here stands, perhaps a different example could be used as this case nor the solution need to use `late`, with the constructor becoming `Point.origin(): this(0, 0);`.

I say this because how I see it, `late` shouldn't be used unless necessary and in this case it's essentially just a hint to the analyzer.

Reply:
I suspect the `late` was just a side effect of migrating the example. But you're right, maybe we should feature it more.

Reply:
Yes, the use of `late` is a side effect of explaining Named Constructors.

The actual good solution is what @parlough  said, but that solution is explained a bit later in the Redirecting Constructors section.

In the codelab there's a line explaining: "This example uses the late annotation to indicate that x and y will be initalized later in the named constructor."

I could probably add a line explaining that we no longer need to annotate with `late` any more because because we are using a redirecting constructor in the other section.

Reply:
On second thought, I'm not sure we want to feature `late` too much, so let's leave it as-is, for now. (We don't want people to use it unnecessarily.)